test/Microsoft.ML.Tests/Scenarios/Api/Metacomponents.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.FastTree;
+using Microsoft.ML.Runtime.Learners;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.ML.Tests.Scenarios.Api
+{
+    public partial class ApiScenariosTests
+    {
+        /// <summary>
+        /// Meta-components: Meta-components (e.g., components that themselves instantiate components) should not be booby-trapped.
+        /// When specifying what trainer OVA should use, a user will be able to specify any binary classifier.
+        /// If they specify a regression or multi-class classifier ideally that should be a compile error.
+        /// </summary>
+        [Fact]
+        void Metacomponents()
+        {
+            var dataPath = GetDataPath(IrisDataPath);
+            using (var env = new TlcEnvironment())
+            {
+                var loader = new TextLoader(env, MakeIrisTextLoaderArgs(), new MultiFileSource(dataPath));
+                var term = new TermTransform(env, loader, "Label");
+                var concat = new ConcatTransform(env, term, "Features", "SepalLength", "SepalWidth", "PetalLength", "PetalWidth");
+                var trainer = new Ova(env, new Ova.Arguments
+                {
+                    PredictorType = new SubComponent<ITrainer<IPredictorProducing<float>>, SignatureBinaryClassifierTrainer>
+                     (
+                         FastTreeBinaryClassificationTrainer.LoadNameValue
+                     )
+                });
+
+                IDataView trainData = trainer.Info.WantCaching ? (IDataView)new CacheDataView(env, concat, prefetch: null) : concat;
+                var trainRoles = new RoleMappedData(trainData, label: "Label", feature: "Features");
+
+                // Auto-normalization.
+                NormalizeTransform.CreateIfNeeded(env, ref trainRoles, trainer);
+                var predictor = trainer.Train(new Runtime.TrainContext(trainRoles));
+
+                var scoreRoles = new RoleMappedData(concat, label: "Label", feature: "Features");
+                IDataScorerTransform scorer = ScoreUtils.GetScorer(predictor, scoreRoles, env, trainRoles.Schema);
+
+                // Cut out term transform from pipeline.
```

Comment:
>// Cut out term transform from pipeline. [](start = 16, length = 40)

let's remove the 'cut out' part and instead use IrisData with label: this will save some lines.

Reply:
How I will map key value to string?

---
In reply to: [208767370](https://github.com/dotnet/machinelearning/pull/653#discussion_r208767370) [](ancestors = 208767370)

Reply:
No need to answer

---
In reply to: [209040525](https://github.com/dotnet/machinelearning/pull/653#discussion_r209040525) [](ancestors = 209040525,208767370)